### ISSUE TITLE
editorconfig binds still show up despite being commented in init.el

### ISSUE THREAD

**Issue by cole-h** [hlissner/doom-emacs#1312]
### Observed behavior

I'm back again! I found this issue by accidentally fat-fingering `SPC f c` which is present even when `editorconfig` is commented out in my private init.el. Running it gives `Wrong type argument: commandp, editorconfig-find-current-editorconfig` (which is to be expected because editorconfig isn't loaded).

### Expected behavior

`editorconfig` and related binds/config settings should be feature-gated (don't know what else to call it when you use the `featurep!` macro to conditionally enable stuff) so that it gets removed when commented in init.el (uncommenting currently works as expected: downloads package, etc).

### Steps to reproduce

(reproducable on both daemon and normal)
1. Launch Doom Emacs
2. `SPC f c`
3. Error

### System information

<details>
<summary>Click to expand</summary>

- OS: gnu/linux (x86_64-pc-linux-gnu)
- Emacs: 26.1 (Mar 17, 2019)
- Doom: 2.0.9 (develop ffe297bc7f49be3dd751b7a5ed1978a350d268ed)
- Graphic display: t (daemon: t)
- System features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES THREADS LIBSYSTEMD LCMS2
- Details:
  ```elisp
  env bootstrapper: envvar-file
  elc count: 0
  uname -a:  Linux Scadrial 5.0.6-arch1-1-custom #1 SMP Fri Apr 5 16:39:33 PDT 2019 x86_64 GNU/Linux
  modules:   (:feature eval (evil +everywhere) file-templates (lookup +docsets) :completion company ivy :ui doom doom-dashboard hl-todo modeline treemacs (popup +all +defaults) vc-gutter vi-tilde-fringe window-select :editor fold multiple-cursors rotate-text :emacs dired electric imenu vc :tools flycheck lsp magit :lang data emacs-lisp lua markdown (org +attach +babel +capture +export +present) (rust +lsp) (sh +zsh) :config default)
  packages:  (move-text aggressive-indent)
  exec-path: (~/bin ~/.local/bin/ ~/.go/bin ~/.cargo/bin /usr/local/bin /usr/bin /bin /usr/local/sbin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.1/x86_64-pc-linux-gnu/)
  ```
</details>

**Comment by hlissner** [comment]
This is an ongoing issue I haven't decided how to deal with. Most keybinds in the `:config default` module are bound unconditionally. I avoid feature-gating them to avoid:

1. the performance impact: `map!` and `general` (the mechanisms through which I define leader keys) are relatively expensive for this purpose -- accounting for ~20% of Doom's startup time -- though I gradually chip away at its overhead.
2. the readability impact: the `:config default` module serves as one big reference for Doom's keybinds. Keybinds are arranged alphabetically and/or grouped by their function, rather than their origin. Feature gating commands will force me to ruin that arrangement and inflate the file's LOC. Documentation could serve as a solution for this, but I haven't gotten around to that (or figured out how to organize such a thing for so many keybinds).

I also avoid moving these keybinds into their respective modules so that keybinding schemes can be modularized (and be toggled if desired).

Long story short, I think `commandp` errors are the lesser evil. 

I've looked into ways to get map! to ignore missing commands or replace them with stubs to emit more helpful errors, but these solutions have their own performance or UX  issues.

**Comment by cole-h** [comment]
For now, enabling the editorconfig section fixes the problem (or rather, avoids it) and there aren't really any downsides to doing so AFAICT, so I will do just that.

**Comment by hlissner** [comment]
Good to hear! In that case, I'll consider this resolved.
